go/interpreter/lfvm/memory_test.go

```diff
@@ -47,3 +49,136 @@ func TestExpansionCosts(t *testing.T) {
 		}
 	}
 }
+
+func TestMemory_expandMemoryWithoutCharging(t *testing.T) {
+
+	test := map[string]struct {
+		size        uint64
+		initialMem  []byte
+		expectedMem []byte
+	}{
+		"empty-memory-increases-to-desired-size": {
+			size:        32,
+			initialMem:  []byte{},
+			expectedMem: []byte{31: 0x0},
+		},
+		"memory-bigger-than-size-changes-nothing": {
+			size:        32,
+			initialMem:  []byte{63: 0x0},
+			expectedMem: []byte{63: 0x0},
+		},
+		"size-zero-changes-nothing": {
+			size:        0,
+			initialMem:  []byte{},
+			expectedMem: []byte{},
+		},
+	}
+
+	for name, test := range test {
+		t.Run(name, func(t *testing.T) {
+			m := NewMemory()
+			fee := m.ExpansionCosts(test.size)
+			m.store = test.initialMem
+			m.expandMemoryWithoutCharging(test.size, fee)
+			if !bytes.Equal(m.store, test.expectedMem) {
+				t.Errorf("unexpected memory value, want: %x, got: %x", test.expectedMem, m.store)
+			}
+			if m.total_memory_cost != fee {
+				t.Errorf("unexpected total memory cost, want: %d, got: %d", fee, m.total_memory_cost)
+			}
+		})
+	}
+
+}
+
+func TestMemory_expandMemory_ErrorCases(t *testing.T) {
+
+	tests := map[string]struct {
+		size     uint64
+		offset   uint64
+		gas      tosca.Gas
+		expected error
+	}{
+		"not enough gas": {
+			size:     32,
+			offset:   0,
+			gas:      0,
+			expected: errOutOfGas,
+		},
+		"offset overflow": {
+			size:     1,
+			offset:   math.MaxUint64,
+			gas:      100,
+			expected: errGasUintOverflow,
+		},
+		"size overflow": {
+			size:     math.MaxUint64,
+			offset:   1,
+			gas:      100,
+			expected: errGasUintOverflow,
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctxt := getEmptyContext()
+			m := NewMemory()
+			ctxt.gas = test.gas
+
+			err := m.expandMemory(test.offset, test.size, &ctxt)
+			if !errors.Is(err, test.expected) {
+				t.Errorf("unexpected error, want: %v, got: %v", test.expected, err)
+			}
+			if m.Len() != 0 {
+				t.Errorf("should have remained empty, got length: %d", m.Len())
+			}
+		})
+	}
+}
+
+func TestMemory_expandMemory(t *testing.T) {
+
+	tests := map[string]struct {
+		size          uint64
+		offset        uint64
+		initialMemory []byte
```

Comment:
The `initialMemory` field is only checked below if `size == 0`, but in all those cases the `initialMemory` field is `nil`. So it does not seem to serve a purpose.